astropy/__init__.py

```diff
@@ -219,6 +219,9 @@ def _initialize_astropy():
     is_astropy_source_dir = os.path.exists(os.path.join(source_dir, os.pardir,
                                                         '.astropy-root'))
 
+    if sys.version_info[:2] < (2,7):
+        warn("Python 2.6 will no longer be supported from Astropy v1.2 onwards.", DeprecationWarning)
```

Comment:
This should probably still be an `AstropyDeprecationWarning`, unless that causes too many problems at import-time.